src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs

```diff
@@ -0,0 +1,421 @@
+            }
+        }
+
+        #endregion Enable and disable geometry buttons methods
+
+        private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            try
+            {
+                if (!_geometryEditor.IsStarted)
+                {
+                    // Identify graphics in the graphics overlay using the mouse point.
+                    IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 2, false);
+
+                    // Get the first graphic if results were found.
+                    IdentifyGraphicsOverlayResult idResult = results.FirstOrDefault();
+                    if (idResult != null && idResult.Graphics.Count > 0)
+                    {
+                        _selectedGraphic = idResult.Graphics.FirstOrDefault();
+                        _selectedGraphic.IsSelected = true;
+
+                        GeometryType geometryType = _selectedGraphic.Geometry.GeometryType;
+
+                        // Configure the UI depending on the geometry type.
+                        if (geometryType == GeometryType.Point || geometryType == GeometryType.Multipoint)
+                        {
+                            ToolComboBox.SelectedIndex = 0;
+                            ToolComboBox.IsEnabled = false;
+                        }
+                        else
+                        {
+                            ToolComboBox.IsEnabled = true;
+                        }
+                        switch (geometryType)
+                        {
+                            case GeometryType.Point:
+                                DisableOtherGeometryButtons(PointButton);
+                                break;
+
+                            case GeometryType.Multipoint:
+                                DisableOtherGeometryButtons(MultipointButton);
+                                break;
+
+                            case GeometryType.Polyline:
+                                DisableOtherGeometryButtons(PolylineButton);
+                                break;
+
+                            case GeometryType.Polygon:
+                                DisableOtherGeometryButtons(PolygonButton);
+                                break;
+                        }
+
+                        // Hide the selected graphic and start an editing session with a copy of it.
+                        _geometryEditor.Start(_selectedGraphic.Geometry);
+                        _selectedGraphic.IsVisible = false;
+                    }
+                    else
+                    {
+                        _selectedGraphic = null;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // Report exceptions.
+                MessageBox.Show("Editing error: " + ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
+            }
```

Comment:
Will you need to restore the selection and visibility of `_selectedGraphic` if there's any on error?

Reply:
I wonder how risky it would be to get rid of this try-catch block. I haven't gotten exceptions after thorough testing all buttons.

Reply:
Disregard previous comment - would need to surround the await code and anything that uses the results in a try/catch block. And yes, used your `ResetSelectedGraphic()` method in this case.